### PR TITLE
Remove 3-d scatter plots support for dense data

### LINKED ISSUE #2046: Remove builtin support for 3-D scatterplots of N-D data
Scipp currently support 3-D scatterplots of N-D data (see first figure in https://scipp.github.io/visualization/plotting/plotting-nd-data.html#3D-visualization). This works by, among other things, computing a regular grid of positions, and dealing with scaling differences of the axis.

We feel that:
1. This is of very limited use, i.e., the number of users who would regularely use this feature is extremely small compared to users of regular plots.
2. Development effort and maintenance overhead is not insignificant.
3. It is relatively simple to manually create fake position and such a plot. We can provide a recipe for this in the docs. Only advanced features such as switching the dims which are mapped to positions or not possible. Example:
   ```python
   xx = da.coords['xx'].copy()
   yy = da.coords['yy'].copy()
   zz = da.coords['zz'].copy()
   yy.unit = ''  # deal with potential different units
   xx *= 3.0  # deal with potential differences in axis scales
   da.coords['dummy-pos'] = sc.geometry.position(xx, yy, zz)
   da.plot(projection='3d', positions='dummy-pos')
   ```

Therefore the plan is to remove this feature.

### PR DESCRIPTION
**Pull request by SimonHeybrock**
Fix #2046.

### CODE REVIEW

**Review by nvaytet — CHANGES_REQUESTED** [review]
Looks good, some small comments

File: docs/visualization/plotting/plotting-nd-data.ipynb [hunk 1]
```diff
@@ -152,17 +151,38 @@
...
+    "y = r * np.sin(theta) * np.cos(phi)\n",
+    "z = r * np.cos(theta)\n",
+    "\n",
+    "a = np.arange(M*N).reshape([M, N]) * np.sin(y)\n",
+    "da = sc.DataArray(\n",
+    "    data=sc.array(dims=['time', 'xyz'], values=a),\n",
+    "    coords={\n",
+    "        'xyz':sc.vectors(dims=['xyz'], unit='m', values=np.array([x, y, z]).T),\n",
+    "        'time':sc.array(dims=['time'], unit='s', values=np.arange(M).astype(float))})\n",
+    "da.plot(projection='3d', positions='xyz')"
```
[the remark this answers is not in the document]
Comment (SimonHeybrock) [reply docs/visualization/plotting/plotting-nd-data.ipynb:169]: `da.plot(scatter='position')` looks ok, but wouldn't it be odd if the `projection` argument wouldn't work in combination with `scatter`?

File: docs/visualization/plotting/plotting-nd-data.ipynb [hunk 2]
```diff
@@ -171,22 +191,31 @@
...
-    "N = 1000\n",
-    "M = 100\n",
-    "theta = np.random.random(N) * np.pi\n",
-    "phi = np.random.random(N) * 2.0 * np.pi\n",
-    "r = 10.0 + (np.random.random(N) - 0.5)\n",
-    "x = r * np.sin(theta) * np.sin(phi)\n",
-    "y = r * np.sin(theta) * np.cos(phi)\n",
-    "z = r * np.cos(theta)\n",
+    "d.coords['dummy-pos'] = sc.geometry.position(*[d.coords[dim] for dim in ['x', 'y', 'z']])\n",
+    "sc.plot(d, projection='3d', positions='dummy-pos')"
```
[the remark this answers is not in the document]
Comment (SimonHeybrock) [reply docs/visualization/plotting/plotting-nd-data.ipynb:195]: Yes, I thought no one can see a 1% opacity difference, but if pixels are overlapping that may not be the case.

File: docs/visualization/plotting/plotting-nd-data.ipynb (hunk 1 above)
[the remark this answers is not in the document]
Comment (nvaytet) [reply docs/visualization/plotting/plotting-nd-data.ipynb:169]: I guess the idea would be to try and remove the `projection` argument altogether?
Not sure what a good way would be to request a 1d plot...

[the remark this answers is not in the document]
Comment (nvaytet) [reply docs/visualization/plotting/plotting-nd-data.ipynb:169]: Should we leave this for now and think of argument names in a later PR?

[the remark this answers is not in the document]
Comment (SimonHeybrock) [reply docs/visualization/plotting/plotting-nd-data.ipynb:169]: I'd say so, it is unrelated.

**Review by nvaytet — APPROVED** [review]

**Pull request merged**